catalog/github/webhook.js

```diff
@@ -7,58 +7,106 @@ var request = require('request');
  *  @param {string} username - github username
  *  @param {string} accessToken - github access token
  *  @param {string} events - list of the events the webhook should fire on
+ *  @param {string} endpoint - The endpoint the openwhisk deployed on
  *  @return {object} whisk async
  */
 function main(params) {
-  var username = params.username;
-  var repository = params.repository;
-  var accessToken = params.accessToken;
+    var username = params.username;
+    var repository = params.repository;
+    var accessToken = params.accessToken;
 
-  var endpoint = 'openwhisk.ng.bluemix.net';
-  var lifecycleEvent = params.lifecycleEvent;
-  var triggerName = params.triggerName.split("/");
+    var endpoint = params.endpoint || 'openwhisk.ng.bluemix.net';
+    var lifecycleEvent = params.lifecycleEvent;
+    var triggerName = params.triggerName.split("/");
 
     // URL of the whisk system. The calls of github will go here.
-  var whiskCallbackUrl = 'https://' + whisk.getAuthKey() + "@" + endpoint + '/api/v1/namespaces/' + triggerName[1] + '/triggers/' + triggerName[2];
+    var whiskCallbackUrl = 'https://' + whisk.getAuthKey() + "@" + endpoint + '/api/v1/namespaces/' + encodeURIComponent(triggerName[1]) + '/triggers/' + encodeURIComponent(triggerName[2]);
 
     // The URL to create the webhook on Github
-  var registrationEndpoint = 'https://api.github.com/repos/' + username + '/' + repository + '/hooks';
+    var registrationEndpoint = 'https://api.github.com/repos/' + username + '/' + repository + '/hooks';
 
-  var authorizationHeader = 'Basic ' + new Buffer(username + ':' +
-  accessToken).toString('base64');
+    var authorizationHeader = 'Basic ' + new Buffer(username + ':' +
+        accessToken).toString('base64');
 
-  if (lifecycleEvent === 'CREATE') {
-    var events = params.events.split(',');
+    if (lifecycleEvent === 'CREATE') {
+        var events = params.events.split(',');
 
-    var body = {
-      name: 'web',
-      active: true,
-      events: events,
-      config: {
-        url: whiskCallbackUrl,
-        content_type: 'json'
```

Comment:
Either return in if block above with return whisk.error or guard next code block with else.